server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1528,16 +1528,21 @@ private NoOpResult innerNoOp(final NoOp noOp) throws IOException {
 
     @Override
     public void refresh(String source) throws EngineException {
-        refresh(source, SearcherScope.EXTERNAL);
+        refresh(source, SearcherScope.EXTERNAL, true);
     }
 
-    final void refresh(String source, SearcherScope scope) throws EngineException {
+    @Override
+    public boolean maybeRefresh(String source) throws EngineException {
+        return refresh(source, SearcherScope.EXTERNAL, false);
```

Comment:
If an INTERNAL refresh is happening, then we skip a scheduled refresh (EXTERNAL scope), then the latest changes won't be visible in the searchers. Is it okay?

Reply:
This is not an issue.